services/resource-service/server/src/test/java/org/apache/airavata/mft/resource/server/backend/file/TestFileBasedResourceBackend.java

```diff
@@ -0,0 +1,273 @@
+package org.apache.airavata.mft.resource.server.backend.file;
+
+import org.apache.airavata.mft.resource.service.*;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class TestFileBasedResourceBackend {
+
+    static FileBasedResourceBackend fileBasedResourceBackend;
+
+    @BeforeAll
+    public static void before() {
+        fileBasedResourceBackend = Mockito.spy(new FileBasedResourceBackend());
+        Mockito.doReturn("resources.json").when(fileBasedResourceBackend).getResourceFile();
+    }
+
+//    @Test
+//    public void testGetFtpResource_WithProperResourceId() {
+//        FileBasedResourceBackend fileBasedResourceBackend = new FileBasedResourceBackend();
+//        String resourceId = "ftp-resource";
+//        FTPResourceGetRequest resourceGetRequest = Mockito.mock(FTPResourceGetRequest.class);
+//        Mockito.when(resourceGetRequest.getResourceId()).thenReturn(resourceId);
+//
+//        try {
+//            Optional<FTPResource> ftpResourceOptional = fileBasedResourceBackend.getFTPResource(resourceGetRequest);
+//            assertTrue(ftpResourceOptional.isPresent());
+//
+//            FTPResource ftpResource = ftpResourceOptional.get();
+//            assertEquals(resourceId, ftpResource.getResourceId());
+//            assertNotNull(ftpResource.getResourcePath());
```

Comment:
Since the contents in resources.json might change, instead of asserting on the content, I'm asserting to make sure the property is not null.